simplesat/sat/assignment_set.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+import six
+
+from collections import OrderedDict
+
+
+class _MISSING(object):
+    pass
+MISSING = _MISSING()
+
+
+class AssignmentSet(object):
+
+    """A collection of literals and their assignments."""
+
+    def __init__(self, assignments=None):
+        # Changelog is a dict of id -> (original value, new value)
+        # FIXME: Verify that we really need ordering here
+        self._data = OrderedDict()
+        self._orig = {}
+        self._cached_changelog = None
+        self._assigned_literals = set()
+        for k, v in (assignments or {}).items():
+            self[k] = v
+
+    def __setitem__(self, key, value):
+        assert key > 0
+
+        prev_value = self.get(key)
+
+        if prev_value is not None:
+            self._assigned_literals.difference_update((key, -key))
+
+        if value is not None:
+            self._assigned_literals.add(key if value else -key)
```

Comment:
Consequently, assuming the above comment is accurate, when does `_assigned_literals` get negative keys ?